plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/* eslint-disable react/sort-prop-types */
+import dt from 'datatables.net-bs';
+import PropTypes from 'prop-types';
+import { formatNumber } from '@superset-ui/number-format';
+import fixTableHeight from './utils/fixTableHeight';
+import 'datatables.net-bs/css/dataTables.bootstrap.css';
+
+if (window.$) {
+  dt(window, window.$);
+}
+const $ = window.$ || dt.$;
+
+const propTypes = {
+  data: PropTypes.shape({
+    // TODO: replace this with raw data in SIP-6
+    html: PropTypes.string,
+    columns: PropTypes.arrayOf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
+    ),
+  }),
+  height: PropTypes.number,
+  columnFormats: PropTypes.objectOf(PropTypes.string),
+  numberFormat: PropTypes.string,
+  numGroups: PropTypes.number,
+  verboseMap: PropTypes.objectOf(PropTypes.string),
+};
+
+function PivotTable(element, props) {
+  const { data, height, columnFormats, numberFormat, numGroups, verboseMap } = props;
+
+  const { html, columns } = data;
+  const container = element;
+  const $container = $(element);
+
+  // queryData data is a string of html with a single table element
+  container.innerHTML = html;
+
+  const cols = Array.isArray(columns[0]) ? columns.map(col => col[0]) : columns;
+
+  // jQuery hack to set verbose names in headers
+  // eslint-disable-next-line func-name-matching
+  const replaceCell = function replace() {
+    const s = $(this)[0].textContent;
+    $(this)[0].textContent = verboseMap[s] || s;
+  };
+  $container.find('thead tr:first th').each(replaceCell);
+  $container.find('thead tr th:first-child').each(replaceCell);
+
+  // jQuery hack to format number
+  $container.find('tbody tr').each(function eachRow() {
+    $(this)
+      .find('td')
+      .each(function each(i) {
+        const metric = cols[i];
+        const format = columnFormats[metric] || numberFormat || '.3s';
+        const tdText = $(this)[0].textContent;
+        if (!Number.isNaN(tdText) && tdText !== '' && tdText.trim().toLowerCase() !== 'null') {
+          $(this)[0].textContent = formatNumber(format, tdText);
+          $(this).attr('data-sort', tdText);
```

Comment:
Note: null values are handled differently in charts. I just changed Big Number to render it as "N/A". Probably makes sense to consolidate the behaviors as we move along refactoring charts.

Reply:
Umm, perhaps we should discuss with Erik before changing that. He was doing work on missing data earlier.